app/controllers/IncomeController.scala

```diff
@@ -110,34 +125,49 @@ class IncomeController @Inject()(personService: PersonService,
           journeyCacheService.collectedValues(Seq(UpdateIncome_PayToDateKey, UpdateIncome_IdKey, UpdateIncome_NameKey), Seq(UpdateIncome_DateKey)) flatMap tupled {
             (mandatorySeq, optionalSeq) => {
               val date = optionalSeq.head.map(date => LocalDate.parse(date))
-              EditIncomeForm.bind(mandatorySeq(2), BigDecimal(mandatorySeq.head), date).fold(
-                formWithErrors => {
+              val employerName = mandatorySeq(2)
+
+              EditIncomeForm.bind(employerName, BigDecimal(mandatorySeq.head), date).fold(
+                (formWithErrors: Form[EditIncomeForm]) => {
                   val webChat = true
                   Future.successful(BadRequest(views.html.incomes.editIncome(formWithErrors,
                     false,
                     mandatorySeq(1).toInt,
                     mandatorySeq.head, webChat = webChat)))
                 },
-                income => {
-                  for {
-                    currentCache <- journeyCacheService.currentCache
-                  } yield {
-                    val newAmount = income.newAmount.getOrElse("0")
-
-                    if (FormHelper.areEqual(currentCache.get(UpdateIncome_ConfirmedNewAmountKey), Some(newAmount))) {
-                      Redirect(routes.IncomeController.sameEstimatedPay())
-                    } else {
-                      journeyCacheService.cache(UpdateIncome_NewAmountKey, newAmount)
-                      Redirect(routes.IncomeController.confirmRegularIncome())
-                    }
-                  }
-                }
+                (income: EditIncomeForm) => determineEditRedirect(income)
               )
             }
           }
         }
   }
 
+  private def determineEditRedirect(income: EditIncomeForm)(implicit hc: HeaderCarrier): Future[Result] = {
+    for {
+      currentCache <- journeyCacheService.currentCache
+    } yield {
+      val newAmount = income.newAmount.getOrElse("0")
```

Comment:
Could maybe inline income.newAmount, and change isCachedIncomeTheSame to take Option[String] since that is what it passes to areEqual. That way in cases of None it would simply return false when comparing to String. My only problem with this is this getOrElse("0") however I can see how further down we need to cache an amount if it's None so we do need the default value of "0". Your thoughts?